File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = {};

        Queue<Integer> queue = new LinkedList<>();

        for (int i = 0; i < progresses.length; i++) {
            int dif = 100 - progresses[i];
            int speed = speeds[i];

            if (dif % speed == 0) {
                queue.add(dif / speed);
            } else {
                queue.add((dif / speed) + 1);
            }
        }

        List<Integer> list = new ArrayList<>();

        while (!queue.isEmpty()) {
            int cnt = 1;

            int work = queue.poll();

            while (!queue.isEmpty() && work >= queue.peek()) {
                cnt++;
                queue.poll();
            }

            list.add(cnt);
        }

        answer = list.stream().mapToInt(Integer::intValue).toArray();

        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    days = 0
    count = 0
    while len(progresses) > 0 :
        if (progresses[0] + days * speeds[0]) >= 100 :
            progresses.pop(0)
            speeds.pop(0)
            count += 1
        else :
            if count > 0 :
                answer.append(count)
                count = 0
            else :
                days += 1
    answer.append(count)
    return answer